bin/cache/update_dart_sdk.sh

```diff
@@ -41,7 +41,15 @@ if [ ! -f "$DART_SDK_STAMP_PATH" ] || [ "$DART_SDK_VERSION" != `cat "$DART_SDK_S
   DART_SDK_ZIP="$FLUTTER_ROOT/bin/cache/dart-sdk.zip"
 
   curl --progress-bar -continue-at=- --location --output "$DART_SDK_ZIP" "$DART_SDK_URL"
-  unzip -o -q "$DART_SDK_ZIP" -d "$FLUTTER_ROOT/bin/cache"
+  unzip -o -q "$DART_SDK_ZIP" -d "$FLUTTER_ROOT/bin/cache" || {
+    echo " "
+    echo "It appears that the downloaded file was corrupted; Please the operation again later."
+    echo "If this problem persists, please report the problem at"
+    echo "https://github.com/flutter/flutter/issues/new"
+    echo " "
+    rm -f -- "$DART_SDK_ZIP"
```

Comment:
since we delete the file regardless, should we delete it before the call to curl as well, just in case the machine hard-crashed during the unzip or something?